File: clog-go/db/entity/dbq-log-entity.go
/*
 * Copyright (c) 2023.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

package entity

import "time"

type DbqLog struct {
	Id         string    `db:"id"`
	Uid        string    `db:"uid"`
	UserId     *string   `db:"user_id"`
	PartnerId  *string   `db:"partner_id"`
	Xid        *string   `db:"xid"`
	SvcName    string    `db:"svc_name"`
	SvcVersion string    `db:"svc_version"`
	SvcParent  *string   `db:"svc_parent"`
	SqlQuery   string    `db:"sql_query"`
	SqlPars    *string   `db:"sql_pars"`
	Severity   string    `db:"severity"`
	Path       string    `db:"path"`
	Function   string    `db:"function"`
	Error      *string   `db:"error"`
	StackTrace *string   `db:"stack_trace"`
	DurationMs int       `db:"duration_ms"`
	StartAt    time.Time `db:"start_at"`
	FinishAt   time.Time `db:"finish_at"`
	CreatedAt  time.Time `db:"created_at"`
}

File: clog-dart/lib/db/entity/dbq_log_entity.dart
/*
 * Copyright (c) 2022.
 * Created by Andy Pangaribuan. All Rights Reserved.
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 */

class DbqLogEntity {
  String id;
  String uid;
  String? userId;
  String? partnerId;
  String? xid;
  String svcName;
  String svcVersion;
  String? svcParent;
  String sqlQuery;
  String? sqlPars;
  String severity;
  String path;
  String function;
  String? error;
  String? stackTrace;
  late int durationMs;
  DateTime startAt;
  DateTime finishAt;
  DateTime createdAt;

  DbqLogEntity({
    required this.id,
    required this.uid,
    this.userId,
    this.partnerId,
    this.xid,
    required this.svcName,
    required this.svcVersion,
    this.svcParent,
    required this.sqlQuery,
    this.sqlPars,
    required this.severity,
    required this.path,
    required this.function,
    this.error,
    this.stackTrace,
    required this.startAt,
    required this.finishAt,
    required this.createdAt,
  }) {
    durationMs = finishAt.difference(startAt).inMilliseconds;
  }
}
